### ISSUE TITLE
WYSIWYG: Saving IFrame-elements directly via TinyMCE editor won't work - IFrame got lost on save

### ISSUE THREAD

**Issue by 4ctobias** [magento/magento2#18550]
### Preconditions
A wysiwyg-editor enabled on the catalog product edit page.  

Tested with Magento 2.2.6

### Steps to reproduce
1. Open the catalog product edit page
2. Open the "content" area"
3. Enable all the WYSIWYG-editors in this area
4. Click on Insert Media in the Description Editor
5. Select Iframe as type
6. Enter a url (eg. https://magento.com/)
7. Click on insert
8. Now save the product

### Expected result

The Iframe is stored within the content and displayed in the frontend, also it will load again when editing the same product again.

### Actual result

The Iframe got lost - it's not shown on the page and also when editing the product again, the Iframe is not longer part of the HTML for the content.

_Workaround for now:_ Hide the editor after inserting the Iframe but before saving, so the HTML will be pushed into a textarea-field - here the transformation is correct and the IFrame got stored as expected.

_Cause_:

When inserting the Iframe with the Media-Dialog, a placeholder img-tag with serialized data is pushed to the preview:

`<img data-mce-src="https://i-18548-2-2-6.engcom.dev.magento.com/pub/static/adminhtml/Magento/backend/en_US/tiny_mce/themes/advanced/img/trans.gif" src="https://i-18548-2-2-6.engcom.dev.magento.com/pub/static/adminhtml/Magento/backend/en_US/tiny_mce/themes/advanced/img/trans.gif" class="mceItemMedia mceItemIframe" data-mce-json="{'type':'iframe','video':{'sources':{}},'params':{'src':'https://magento.com/'},'width':'320','height':'240'}" width="320" height="240">`

On saving the img-tag must be serialized with the TinyMCE serializer, so the data in the data-mce-json-attribute will be transformed to an valid iframe-tag. This is not done at the moment, due to the following call in the tiny_mce_src.js within the function getContent():

```
// Get raw contents or by default the cleaned contents
if (args.format == 'raw')
    content = self.getBody().innerHTML;
else
    content = self.serializer.serialize(self.getBody(), args);
```

As the getContent is called with format "raw" from the onBeforeUnload-Handler as defined also in tiny_mce_src.js the iframe information got lost, as the serializer got not called:

````
if (s.add_unload_trigger) {
    t._beforeUnload = tinyMCE.onBeforeUnload.add(function() {
        if (t.initialized && !t.destroyed && !t.isHidden())
	    t.save({format : 'raw', no_events : true});
    });
}
´´´

**Comment by magento-engcom-team** [comment]
Hi @4ctobias. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@4ctobias do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by 4ctobias** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @4ctobias. Thank you for your request. I'm working on Magento 2.2.6 instance for you

**Comment by magento-engcom-team** [comment]
Hi @4ctobias, here is your [Magento instance](https://i-18550-2-2-6.engcom.dev.magento.com).
Admin access: [https://i-18550-2-2-6.engcom.dev.magento.com/admin](https://i-18550-2-2-6.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by 4ctobias** [comment]
@magento-engcom-team Yes, I was able to reproduce the issue on the vanilla Magento instance.

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by eduard13** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @eduard13, here is your [Magento instance](http://34.228.235.121/i-18550-2-3-develop/).
Admin access: [http://34.228.235.121/i-18550-2-3-develop//admin](http://34.228.235.121/i-18550-2-3-develop//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
@eduard13 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96160` were created

**Comment by engcom-Charlie** [comment]
Hello @4ctobias 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the lates `2.3-develop` branch by provided steps.

We are closing this issue due to _branch `2.2-develop` was closed and Magento no longer accepts pull requests and issues to the v2.2 release lines to focus all development efforts on v2.3._
 See [Accepted pull requests and ported code](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules) for more details

If you still faced this issue on `2.3` please create a new Issue with all required details according to [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)

**Comment by juretopolak** [comment]
@engcom-Charlie I have exactly the same problem with Magento 1.9.4.4.

I was just searching for commits in which this issue was solved but couldn't find it.

Would be someone so kind and help me find it, maybe I could somehow help myself with it.

**Comment by denweng** [comment]
@engcom-Charlie we also have the Problem in 1.9.3 some hint which changes did the fix would be great.

**Comment by juretopolak** [comment]
@denweng here is described how I solved same problem with Magento 1.9.4.4:

https://magento.stackexchange.com/questions/311112/iframe-got-lost-on-save-using-wysiwyg-tinymce-magento-1-9-4-4

Maybe it can help you.

**Comment by denweng** [comment]
A big THANKX, it works, you saved my day!

**Comment by MagePsycho** [comment]
What's the solution for this?
